### ISSUE TITLE
Duplicate ATHS file content will crash Tor service on Tails

### ISSUE THREAD

**Issue by msheiny** [freedomofpress/securedrop#1976]
# Bug

## Description

This is kind of a convoluted issue - if an admin coming from a pre 0.4 release -- carries over their persistent storage from tails, updates the repo to the latest release, then tries to re-run the `securedrop-admin install` they could end up a scenario with two aths files with the same content. When the `tailsconfig` gets re-run the duplicate `HidServAuth`  lines get copied into /etc/tor/torrc and this causes the tor service to die upon restart.

## Steps to Reproduce

* Have an existing securedrop 0.3.12 installation and tails stick
* Upgrade that tails stick to 3.0.1 and checkout the `release/0.4` branch in ~/Persistent/securedrop. There should still be an existing file for the old journalist interface naming convention (ie `app-document-aths` )
* Run through ./securedrop-admin, `setup`, `sdconfig`, `install`, `tailsconfig`
* Restart the tails stick. Tor should be inaccessible once our network hook (~/Persistent/.securedrop/securedrop_init.py) is called.

## Expected Behavior

There should be no errors when running through this process and tor should continue to work as expected on the tails admin workstation.

## Actual Behavior

Tor will crash with the following entries in the error logs:

```
Jul 13 14:35:51 amnesia systemd[1]: Starting Anonymizing overlay network for TCP...
Jul 13 14:35:51 amnesia tor[13039]: Jul 13 14:35:51.355 [notice] Tor 0.3.0.9 (git-100816d92ab5664d) running on Linux with Libevent 2.0.21-stable, OpenSSL 1.1.0f and Zlib 1.2.8.
Jul 13 14:35:51 amnesia tor[13039]: Jul 13 14:35:51.355 [notice] Tor can't help you if you use it wrong! Learn how to be safe at https://www.torproject.org/download/download#warning
Jul 13 14:35:51 amnesia tor[13039]: Jul 13 14:35:51.355 [notice] Read configuration file "/usr/share/tor/tor-service-defaults-torrc".
Jul 13 14:35:51 amnesia tor[13039]: Jul 13 14:35:51.356 [notice] Read configuration file "/etc/tor/torrc".
Jul 13 14:35:51 amnesia tor[13039]: Jul 13 14:35:51.358 [warn] The ControlListenAddress option is deprecated, and will most likely be removed in a future version of Tor. Use ControlPort instead.
Jul 13 14:35:51 amnesia tor[13039]: Jul 13 14:35:51.358 [warn] The TransListenAddress option is deprecated, and will most likely be removed in a future version of Tor. Use TransPort instead. (If
Jul 13 14:35:51 amnesia tor[13039]: Jul 13 14:35:51.358 [warn] The WarnUnsafeSocks option is deprecated, and will most likely be removed in a future version of Tor. Changing this option makes it
Jul 13 14:35:51 amnesia tor[13039]: Jul 13 14:35:51.358 [warn] Duplicate authorization for the same hidden service.
Jul 13 14:35:51 amnesia tor[13039]: Jul 13 14:35:51.359 [warn] Failed to parse/validate config: Failed to configure client authorization for hidden services. See logs for details.
Jul 13 14:35:51 amnesia tor[13039]: Jul 13 14:35:51.359 [err] Reading config failed--see warnings above.
Jul 13 14:35:51 amnesia systemd[1]: tor@default.service: Control process exited, code=exited status=1
Jul 13 14:35:51 amnesia systemd[1]: Failed to start Anonymizing overlay network for TCP.
```

## Comments

Note there are multiple warnings, including some deprecation warnings for other log items we didnt add. I have confirmed that once I removed a duplicate `HidServAuth` line and restarted `tor@default.service` - tor is coming up again.

Suggestion to fix is to tweak `install_files/ansible-base/roles/tails-config/tasks/configure_torrc_additions.yml` so that either:

* We use something like `lineinfile` and lookups to ensure duplicate entries dont get dumped into the torrc additions file. Orrrr
* We more explicitly define the file names that are piped to the `assemble` module.

Suggestions to fix, any other relevant information.

**Comment by conorsch** [comment]
This issue is closely related to #1951, in that the Document -> Journalist Interface renaming is not well handled on the Admin Workstation (see #1614 for context). Rather than resolve individual occurrences of filename conflicts as they arise in QA, we should take a step back and try to resolve the problem cleanly just once.

I propose adding a migration strategy, via renaming the `app-document-aths` file to `app-journalist-aths`, to the `securedrop-admin tailsconfig` entry point. This will give us a single location to handle the update in the context of the Admin (and Journalist) Workstation, similar to how we handled the migration in a single location on the Application Server via a preinst hook in #1650.

In addition to writing the migration logic, we should also explicitly document the need for Admins to run `./securedrop-admin tailsconfig` on the new Tails 3 workstations as part of the 0.4 release. @freddymartinez9 pointed out the need for such Admin-centric documentation in https://github.com/freedomofpress/securedrop/issues/1551#issuecomment-314818161, but we don't have an explicit issue tracking those docs changes yet. I'll open that presently.

**Comment by conorsch** [comment]
Resolved via #2000.
